lldb/test/API/lang/swift/async/stepping/step-in/task-switch/TestSwiftTaskSwitch.py

```diff
@@ -15,11 +15,25 @@ def test(self):
         target, _, thread, _ = lldbutil.run_to_source_breakpoint(self, "await f()", src)
         self.assertEqual(thread.frame[0].function.mangled, "$s1a5entryO4mainyyYaFZ")
 
-        function = target.FindFunctions("$s1a5entryO4mainyyYaFZTQ0_")[0].function
-        instructions = list(function.GetInstructions(target))
+        sym_ctx_list = target.FindFunctions("$s1a5entryO4mainyyYaFZTQ0_", lldb.eFunctionNameTypeAuto)
+        self.assertEquals(sym_ctx_list.GetSize(), 1)
+        symbolContext = sym_ctx_list.GetContextAtIndex(0)
+        function = symbolContext.GetFunction()
+        if function:
+            instructions = list(function.GetInstructions(target))
+        else:
+            symbol = symbolContext.GetSymbol()
+            self.assertIsNotNone(symbol)
+            instructions = list(symbol.GetInstructions(target))
 
         # Expected to be a trampoline that tail calls `swift_task_switch`.
-        self.assertIn("swift_task_switch", instructions[-1].GetComment(target))
+        for inst in reversed(instructions):
+            control_flow_kind = inst.GetControlFlowKind(target)
+            if control_flow_kind == lldb.eInstructionControlFlowKindOther:
+                continue
```

Comment:
can you say why some instructions need to be skipped? Padding instructions?

Reply:
Yes. Getting from the symbol will include the padding directive at the end.
But I'm not clear on lldb's difference between GetSymbol and GetFunction.